codegen/masm/src/codegen.rs

```diff
@@ -114,15 +111,40 @@ fn quadratic_element_square(writer: &mut Writer, n: u32) {
 
 impl<'ast> CodeGenerator<'ast> {
     pub fn new(ir: &'ast AirIR, config: CodegenConfig) -> CodeGenerator<'ast> {
+        // remove duplicates and sort period lengths in descending order, since larger periods will
+        // give smaller number of cycles (which means a smaller number of exponentiations)
```

Comment:
minor nit: "will have smaller" sounds better to me

Reply:
changed